### PR TITLE
`ACE_Time_Value` Should Accept Floating-point-based `std::chrono::duration`

### PR DESCRIPTION
**Pull request by iguessthislldo**
`ACE_Time_Value` currently doesn't accept a `std::chrono::duration` with `Rep` that's a floating point type because it's using the modulus operator to separate the microseconds (`duration % seconds(1)`) and trying to use modulus with floating points causes a compile error.

This changes it to use subtraction from the whole number of seconds (`duration - sec`) and adds a test for it.

## Summary by CodeRabbit

- New Features
  - Support for constructing time values from floating-point std::chrono durations.

- Bug Fixes
  - More consistent sub-second extraction; conversion semantics adjusted (may affect negative-duration edge cases).
  - Internal arithmetic for adding/subtracting durations simplified to reuse construction logic.

- Tests
  - Expanded chrono tests (positive/negative and floating-point cases) and added a reusable test helper.

- Chores
  - Updated test ignore list to reflect current test alignment.

### CODE REVIEW

File: ACE/ace/Time_Value.h [hunk 1]
```diff
@@ -118,13 +118,11 @@ class ACE_Export ACE_Time_Value
...
+    using namespace std::chrono;
 
-    std::chrono::microseconds const usec {
-      std::chrono::duration_cast<std::chrono::microseconds>(
-        duration % std::chrono::seconds (1))};
-    this->set (ACE_Utils::truncate_cast<time_t>(s.count ()), ACE_Utils::truncate_cast<suseconds_t>(usec.count ()));
+    seconds const sec {duration_cast<seconds> (duration)};
+    microseconds const usec {duration_cast<microseconds> (duration  - sec)};
+    this->set (ACE_Utils::truncate_cast<time_t> (sec.count ()), ACE_Utils::truncate_cast<suseconds_t> (usec.count ()));
   }
```
[the remark this answers is not in the document]
Comment (iguessthislldo) [reply ACE/ace/Time_Value.h]: `set` does normalize, so I'm not interested in adding useless code.

File: ACE/tests/Chrono_Test.cpp [hunk 2]
```diff
@@ -156,36 +98,16 @@ test_assignments ()
...
-
-  tv = ACE_Time_Value {usec};
-
   //                       half a day  3 hours   24 minutes 54 seconds
   time_t expected_sec = { ((12*3600) + (3*3600) + (24*60) + 54 ) * 2 };
   //                              238 milli    usec  342 nano
   suseconds_t expected_usec = { ((238*1000) + 754 + 0) * 2 };
+  errors += tv_test_case(usec,
+    "two times half a day, 3 hours, 24 minutes, 54 seconds, "
+    "238 milliseconds, 754 microseconds and 342 nanoseconds",
```
[the remark this answers is not in the document]
Comment (iguessthislldo) [reply ACE/tests/Chrono_Test.cpp:107]: I copied this into a Python `timedelta` and it looks correct, even though the whole thing is a bit convoluted:

```
In [10]: print(timedelta(seconds=((12*3600) + (3*3600) + (24*60) + 54 ) * 2, microseconds=((238*1000) + 754 + 0
    ...: ) * 2) / 2)
15:24:54.238754
```

I did notice after I copied this into GitHub that nanosecond part isn't real, is that what you're talking about?

[the remark this answers is not in the document]
Comment (mitza-oci) [reply ACE/tests/Chrono_Test.cpp:107]: Two things seem off: the comment on line 103 is "238 milli    usec  342 nano" so it's missing the 754 (and shouldn't have nanoseconds since suseconds_t doesn't have any) and the string on line 107 refers to nanoseconds.  I guess the latter is OK if it means that the initial object had 342 (actually 684) nanoseconds and they were truncated.  Is truncation instead of rounding the expected behavior?

[the remark this answers is not in the document]
Comment (iguessthislldo) [reply ACE/tests/Chrono_Test.cpp:107]: I also realized the 342 nanoseconds are lost because they're putting all this in a `microseond` type, so it has absolutely nothing to do with `ACE_Time_Value`.

[the remark this answers is not in the document]
Comment (iguessthislldo) [reply ACE/tests/Chrono_Test.cpp:107]: I missed your last comment, but came to the same conclusion. I decided to change the input type above to `nanoseconds`, that way it's actually giving the nanoseconds to `ACE_Time_Value` so the test makes more sense. I also fixed the comments to make it clear the nanoseconds are lost.

> Is truncation instead of rounding the expected behavior?

It appears to be: https://stackoverflow.com/questions/36122428/how-does-duration-cast-round
